modules/installation-osp-kuryr-known-limitations.adoc

```diff
@@ -5,20 +5,58 @@
-that OpenShift UDP Services are not supported.
-* There is a known limitation of Octavia not supporting listeners on different
-protocols, like UDP and TCP, on the same port. Thus, Services exposing the same
-port for different protocols are not supported.
-* Due to the above UDP limitations of Octavia, Kuryr forces Pods to use TCP
-for DNS resolution. This is set with the `use-vc` option in `resolv.conf`. This
-might be a problem for Pods running Go applications compiled with the `CGO_ENABLED`
-flag disabled, as that uses the `go` resolver that only leverages UDP and is not
-considering the `use-vc` option added by Kuryr to the `resolv.conf`. This is a
-problem also for musl-based containers as its resolver does not support the
-`use-vc` option. This includes images built from `alpine`.
+There are a number of known limitations when using Kuryr SDN.
+
+[discrete]
+[id="openstack-resource-limitations_{context}"]
+== {rh-openstack} resource limitations
+
+* An Amphora load balancer VM is deployed per OpenShift Service that uses the
+default Octavia load balancer driver (Amphora driver). Creating too many Services
+can cause you to run out of resources.
+
+[discrete]
+[id="openstack-version-limitations_{context}"]
+== {rh-openstack} version limitations
+
+There are limitations when using Kuryr SDN that depend on the {rh-openstack}
+version.
+
+* Octavia {rh-openstack} versions before 16 do not support UDP listeners. Therefore,
+OpenShift UDP services are not supported.
+
+* Octavia {rh-openstack} versions before 16 cannot listen to multiple protocols on the
+same port. Services that expose the same port to different protocols, like TCP
+and UDP, are not supported.
+
+[IMPORTANT]
+====
+The OVN Octavia driver does not support listeners that use different protocols on
+any {rh-openstack} version.
+====
+
+[discrete]
+[id="openstack-go-limitations_{context}"]
+== {rh-openstack} environment limitations
+
+There are limitations when using Kuryr SDN that depend on your deployment environment.
+
+* Because of Octavia's lack of support for the UDP protocol and multiple listeners, Kuryr forces Pods to use TCP
+for DNS resolution if:
+** The {rh-openstack} version is earlier than 16
+** The OVN Octavia driver is used
+
+The `use-vc` option defined in each Pod's `resolv.conf` file controls whether TCP is forced for DNS resolution.
+
+In Go versions 1.12 and earlier, applications that are compiled with CGO support disabled use UDP only. In this case,
+the native Go resolver does not recognize the `use-vc` option in `resolv.conf`. As a result, UDP is still used for DNS resolution.
+
+To ensure that TCP forcing is allowed, compile applications either with the environment variable `CGO_ENABLED` set to `1`, i.e. `CGO_ENABLED=1`, or ensure that the variable is absent.
+
+In Go versions 1.13 and later, TCP is used automatically if UDP fails.
+
+[INFO]
+====
+musl-based containers, including Alpine-based containers, do not support the `use-vc` option.
```

Comment:
What's the end result here? Do you have to compile using Go 1.13? Why don't you just make using Go 1.13 a requirement?

Reply:
@luis5tb This is a fair point. Why not just update the prereqs? We can get rid of much/all of this if we do that, right?

Reply:
Still, it is a limitation and the use-vc is needed. I would prefer to leave this section as there may be other things (perhaps in other languages) doing similar things, and having the GO example here will help to understand if the problem is due to that